agent/pkg/crane/crane.go

```diff
@@ -31,10 +31,6 @@ func Serve(cfg *config.Configuration) {
 	grpcToken := cfg.GrpcToken
 	grpcInsecure := cfg.GrpcInsecure
 
-	if grpcToken == "" {
```

Comment:
This is safe to remove, only if in `config.go` it is marked as required.

Reply:
I make validation for grpcToken to not be an empty string 

```
if unvalidatedToken == "" {
		return fmt.Errorf("grpc jwt token can not be unset")
	}
```

after that the struct is being created, so grpcToken won't be "" in any case, isn't it?

Reply:
It is correct. That check too could be removed, if the required option is set.
Consult the docs here https://github.com/ilyakaznacheev/cleanenv `env-required`
The error message is mixed anyway, the type is for JWT token, it has nothing to do with gRPC directly, for this reason I prefer that solution, however I cannot recall how good are those error messages provided by the cleanenv package. Let's see how is it that way.

Reply:
in their doc they show almost the same [example when field is of type custom struct](https://github.com/ilyakaznacheev/cleanenv#custom-value-setter)

also [from their examples with the code ](https://github.com/ilyakaznacheev/cleanenv/blob/35edfee562627f73280939ac560966bcdd439cda/example/custom_setter/main.go#L39)

env-required is not compatible with custom structs. It doesn't know how to transform env value to custom struct (but the setter knows)

i've changed error messages. maybe it can work out that way?